server/src/main/java/org/opensearch/cluster/coordination/RemotePublishRequest.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.coordination;
+
+import java.io.IOException;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+
+public class RemotePublishRequest extends TermVersionRequest {
```

Comment:
I find the `TermVersionRequest` to be rather vague. Term and Version is incomplete without a concrete cluster uuid?

Reply:
Can you extend ClusterStateTermVersion instead? Also cluster manager should send the path for remote manifest to data nodes. This will ensure consistency incase file paths are changed or anything. Cluster manager will remain authority.

Reply:
ClusterStateTermVersion cannot be extended since this is a simple POJO and not a TransportRequest.
TermVersionRequest is currently being extended by ApplyCommitRequest. So following the same convention.